### PR TITLE
NonNullableFieldIsUninitializedSuppressor didn't work when there was a constructor

### LINKED ISSUE #363: NonNullableFieldIsUninitializedSuppressor fails when there is a constructor
The rule works find when there is no constructor at all, as in most NUnit tests.
It fails when there is one, with arguments from a `TestFixtureAttribute` which does not set all non-nullable fields.

When there is no constructor, the error is reported against the field.
When there is a constructor, the error is reported against the constructor.

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #363

### CODE REVIEW

**Review by mikkelbu — APPROVED** [review]
Looks good - although I agree that the extraction of the field name is a bit "hacky". I took a look at the Roslyn source code and found that the error messages are localized, so the solution will not work for e.g. German - 
https://github.com/dotnet/roslyn/blob/64d1beefeca4a0f869cc8fb14fc903af676dcbe4/src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf#L2916 - or Polish - https://github.com/dotnet/roslyn/blob/64d1beefeca4a0f869cc8fb14fc903af676dcbe4/src/Compilers/CSharp/Portable/xlf/CSharpResources.pl.xlf#L2916, but it will work for e.g. French and Japanese.

I see different solutions:
* We use the current implementation until somebody complaints about it not working in a given language, and then we improve on the solution
* We improve the solution now.

I'm happy with both solutions.

Ps. Regarding alternative solutions we could use reflection to extract the field name (I've tried to see, if we can find the name in another way, but so far I've not been successful).

**Comment by manfred-brands** [comment]
I added other separators to the Split so it works for all languages, except Czech which seems to use none.
So, I resorted to reflection getting the 'Arguments[1]' with enough checks to ensure it is a non-null string and also verifying that the string obtained is actually a field name.
If you are happy with reflection only, I will remove the string hack before final merge.

**Review by mikkelbu — APPROVED** [review]
Great work. I think the reflection solution is sufficient (and hopefully MS will not change the arguments in the (near) future).

File: src/nunit.analyzers.tests/DiagnosticSuppressors/NonNullableFieldIsUninitializedSuppressorTests.cs [hunk 1]
```diff
@@ -76,5 +78,48 @@ public async Task FieldAssignedUsingExpressionBody(string attribute, string pref
...
+            {
+                Thread.CurrentThread.CurrentUICulture = CultureInfo.CreateSpecificCulture("cs-CS");
+                await DiagnosticsSuppressorAnalyzer.EnsureSuppressed(suppressor,
+                    NonNullableFieldIsUninitializedSuppressor.NullableFieldInitializedInSetUp, testCode)
+                    .ConfigureAwait(false);
+            }
+            finally
+            {
+                Thread.CurrentThread.CurrentUICulture = savedCulture;
+            }
```
[the remark this answers is not in the document]
Comment (manfred-brands) [reply src/nunit.analyzers.tests/DiagnosticSuppressors/NonNullableFieldIsUninitializedSuppressorTests.cs]: Thanks for the tip, hadn't found that attribute yet.

**Comment by manfred-brands** [comment]
Removed the string extraction, ready for merge.

**Comment by mikkelbu** [comment]
Thanks for another contribution @manfred-brands 👍

**Pull request merged**